lightly/models/modules/center.py

```diff
@@ -0,0 +1,94 @@
+from typing import Callable, Tuple
+
+import torch
+import torch.distributed as dist
+from torch import Tensor
+from torch.nn import Module
+
+
+class Center(Module):
+    """Center module to compute and store the center of a feature tensor as used
+    in DINO [0].
+
+    - [0]: DINO, 2021, https://arxiv.org/abs/2104.14294
+
+    Attributes:
+        size:
+            Size of the tracked center tensor. Dimensions across which the center
+            is computed must be set to 1. For example, if the feature tensor has shape
+            (batch_size, sequence_length, feature_dim) and the center should be computed
+            across the batch and sequence dimensions, the size should be
+            (1, 1, feature_dim).
+        mode:
+            Mode to compute the center. Currently only 'mean' is supported.
+        momentum:
+            Momentum term for the center calculation.
+        _register_buffer:
+            Deprecated, do not use. This argument is only kept for backwards
+            compatibility with DINOLoss.
```

Comment:
Furthermore:
- why is this True by default?
- why is the argument not ignored, but actually used?

Reply:
> How about adding deprecation warning when the argument is used?

Then everyone using the `DINOLoss` would get a deprecation warning.

> why is this True by default?

Because you always want to register the center as a buffer. Otherwise it will not be stored in the state dict.

> why is the argument not ignored, but actually used?

Because it is used for `DINOLoss`

Reply:
This already happens in the DINOLoss itself, it is not something the user has to configure, see: https://github.com/lightly-ai/lightly/blob/b71e1f57ff3f588c3d82d2db4d1e87efecd932c3/lightly/loss/dino_loss.py#L77